tests/components/participants/hooks/useParticipant.test.ts

```diff
@@ -0,0 +1,51 @@
+import { act, renderHook, waitFor } from '@testing-library/react'; // For testing hooks
+
+import { getParticipantByLegalName } from '../../../../src/components/participants/helpers/participantDataFlow'; // Import the mock
+import { useParticipant } from '../../../../src/components/participants/hooks/useParticipant';
+
+import { mockParticipant } from './__fixtures__/participant';
+
+jest.mock('../../../../src/components/participants/helpers/participantDataFlow', () => ({
+  getParticipantByLegalName: jest.fn(),
+}));
+
+describe('useParticipant', () => {
+  const mockLegalName = 'msg systems ag'; // Example legal name for testing
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should return LOADING initially', async () => {
+    (getParticipantByLegalName as jest.Mock).mockReturnValue(new Promise(() => {
+    }));
+
+    const { result } = renderHook(() => useParticipant(mockLegalName));
+
+    await act(async () => {
```

Comment:
Please remove this test. I think it's just pure luck that you could test inbetween state of the application. I think this test is unrilayable. You still get 100% coverage.

Reply:
✓ solved